internal/supervisor/installation.go

```diff
@@ -796,7 +798,10 @@ func (s *InstallationSupervisor) updateInstallation(installation *model.Installa
 
 	logger.Info("Finished updating clusters installations")
 
-	return s.waitForUpdateStable(installation, instanceID, logger)
+	state := s.waitForUpdateStable(installation, instanceID, logger)
+	prometheusTimer.ObserveDuration()
```

Comment:
Unfortunately, this doesn't hold up when we are running multiple provisioners in HA. One provisioner could start the installation update and another one could finish it. Also, I am pretty sure multiple installation updates all occurring at the same time would make this metric incorrect.

Reply:
hmmm I see, the same occurs for creation too right? Or we lock the creation to the provision which started it?

Reply:
Yes it does.

The lock works in a way that prevents other Provisioners to modify Installation when other instance is already working on it but most likely during one lock the given Provisioner will do only some part of work and release the lock (as for example Operator did not finish spinning up Pods yet or something else is not ready). Then there is no guarantee that different Provisioner won't pick it up and perform next step.

Reply:
I am asking because I see we use almost the same logic for creation metrics but I will figure it out how we can change this and be safe for HA

Reply:
@gabrieljackson ^